### ISSUE TITLE
[FreeBSD] - Linker command failure (unable to find library -lemu2149)

### ISSUE THREAD

**Issue by ehaupt** [BambooTracker/BambooTracker#453]
## Bug Description

Trying to compile the newly released [v0.5.0](https://github.com/BambooTracker/BambooTracker/releases/download/v0.5.0/BambooTracker-src-v0.5.0.tar.gz) source tarball containing all the git submodules at the time of the tagging on FreeBSD, I get the following compiler error.

Last command including compiler error:
```
c++ -fstack-protector-strong -pthread -Wl,-rpath,/usr/local/lib/qt5 -o BambooTracker fmopn.o  mame_2608.o  ymdeltat.o  nuked_2608.o  register_write_logger.o  swap_instruments_command.o  change_values_in_pattern_command.o  paste_insert_copied_data_to_pattern_command.o  pattern_command_utils.o  transpose_note_in_pattern_command.o  bookmark_manager_form.o  instrument_command_qt_utils.o  swap_instruments_qt_command.o  order_list_common_qt_command.o  pattern_editor_common_qt_command.o  drop_detect_list_widget.o  effect_description.o  effect_list_dialog.o  file_io_error_message_box.o  go_to_dialog.o  gui_utils.o  hide_tracks_dialog.o  adpcm_sample_editor.o  arpeggio_macro_editor.o  grid_settings_dialog.o  instrument_editor_adpcm_form.o  instrument_editor_drumkit_form.o  instrument_editor_utils.o  pan_macro_editor.o  sample_length_dialog.o  tone_noise_macro_editor.o  key_signature_manager_form.o  keyboard_shortcut_list_dialog.o  note_name_manager.o  swap_tracks_dialog.o  transpose_song_dialog.o  sample_adpcm.o  sequence_property.o  btb_io.o  bti_io.o  btm_io.o  dat_io.o  dmp_io.o  export_io.o  ff_io.o  ins_io.o  io_utils.o  opni_io.o  p86_io.o  ppc_io.o  pps_io.o  pvi_io.o  pzi_io.o  tfi_io.o  vgi_io.o  wav_container.o  wopn_io.o  y12_io.o  jamming.o  main.o  mainwindow.o  chip.o  opna.o  resampler.o  ym3438.o  bamboo_tracker.o  effect.o  note.o  playback.o  song_length_calculator.o  audio_stream.o  instruments_manager.o  command_manager.o  add_instrument_command.o  remove_instrument_command.o  add_instrument_qt_command.o  remove_instrument_qt_command.o  instrument_editor_fm_form.o  fm_operator_table.o  labeled_vertical_slider.o  labeled_horizontal_slider.o  slider_style.o  change_instrument_name_qt_command.o  change_instrument_name_command.o  opna_controller.o  instrument.o  envelope_fm.o  event_guard.o  audio_stream_rtaudio.o  tick_counter.o  module.o  song.o  pattern.o  track.o  step.o  order_list_panel.o  order_list_editor.o  pattern_editor_panel.o  pattern_editor.o  instrument_editor_ssg_form.o  set_key_off_to_step_command.o  set_key_on_to_step_command.o  set_instrument_to_step_command.o  erase_instrument_in_step_command.o  set_volume_to_step_command.o  erase_volume_in_step_command.o  set_effect_id_to_step_command.o  erase_effect_in_step_command.o  set_effect_value_to_step_command.o  erase_effect_value_in_step_command.o  insert_step_command.o  delete_previous_step_command.o  erase_step_command.o  deep_clone_instrument_qt_command.o  deep_clone_instrument_command.o  clone_instrument_command.o  clone_instrument_qt_command.o  set_pattern_to_order_command.o  insert_order_below_command.o  delete_order_command.o  paste_copied_data_to_pattern_command.o  erase_cells_in_pattern_command.o  paste_copied_data_to_order_command.o  instrument_form_manager.o  lfo_fm.o  visualized_instrument_macro_editor.o  effect_iterator.o  paste_mix_copied_data_to_pattern_command.o  module_properties_dialog.o  groove_settings_dialog.o  configuration_dialog.o  expand_pattern_command.o  shrink_pattern_command.o  abstract_instrument_property.o  duplicate_order_command.o  move_order_command.o  clone_patterns_command.o  clone_order_command.o  set_echo_buffer_access_command.o  comment_edit_dialog.o  binary_container.o  interpolate_pattern_command.o  reverse_pattern_command.o  replace_instrument_in_pattern_command.o  vgm_export_settings_dialog.o  wave_export_settings_dialog.o  configuration.o  configuration_handler.o  color_palette.o  paste_overwrite_copied_data_to_pattern_command.o  wopn_file.o  bank.o  instrument_selection_dialog.o  s98_export_settings_dialog.o  precise_timer.o  module_io.o  instrument_io.o  bank_io.o  fm_envelope_set_edit_dialog.o  file_history.o  midi.o  q_application_wrapper.o  wave_visual.o  qrc_doc.o  qrc_icon.o  moc_bookmark_manager_form.o  moc_drop_detect_list_widget.o  moc_effect_list_dialog.o  moc_file_io_error_message_box.o  moc_go_to_dialog.o  moc_hide_tracks_dialog.o  moc_adpcm_sample_editor.o  moc_arpeggio_macro_editor.o  moc_grid_settings_dialog.o  moc_instrument_editor_adpcm_form.o  moc_instrument_editor_drumkit_form.o  moc_pan_macro_editor.o  moc_sample_length_dialog.o  moc_tone_noise_macro_editor.o  moc_key_signature_manager_form.o  moc_keyboard_shortcut_list_dialog.o  moc_mainwindow.o  moc_swap_tracks_dialog.o  moc_transpose_song_dialog.o  moc_audio_stream.o  moc_instrument_editor_fm_form.o  moc_fm_operator_table.o  moc_labeled_vertical_slider.o  moc_labeled_horizontal_slider.o  moc_order_list_panel.o  moc_order_list_editor.o  moc_pattern_editor_panel.o  moc_pattern_editor.o  moc_instrument_editor_ssg_form.o  moc_instrument_form_manager.o  moc_visualized_instrument_macro_editor.o  moc_module_properties_dialog.o  moc_groove_settings_dialog.o  moc_configuration_dialog.o  moc_comment_edit_dialog.o  moc_vgm_export_settings_dialog.o  moc_wave_export_settings_dialog.o  moc_instrument_selection_dialog.o  moc_s98_export_settings_dialog.o  moc_fm_envelope_set_edit_dialog.o  moc_wave_visual.o   -L/usr/local/lib -L/usr/ports/audio/bambootracker/work/BambooTracker-src-v0.5.0/BambooTracker/../submodules/emu2149 -lemu2149 -L/usr/ports/audio/bambootracker/work/BambooTracker-src-v0.5.0/BambooTracker/../submodules/RtAudio -lrtaudio -L/usr/ports/audio/bambootracker/work/BambooTracker-src-v0.5.0/BambooTracker/../submodules/RtMidi -lrtmidi /usr/local/lib/qt5/libQt5Widgets.so /usr/local/lib/qt5/libQt5Gui.so /usr/local/lib/qt5/libQt5Core.so -lGL
ld: error: unable to find library -lemu2149
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

The full build log from the [poudriere](https://www.freebsd.org/cgi/man.cgi?poudriere) build sandbox (FreeBSD ports build sandbox). Can be found [here](http://snow.critical.ch/poudriere/data/130amd64-default/2022-05-28_11h26m15s/logs/errors/bambootracker-0.5.0.log).

As the issue checklist states, I confirmed that the same error occurs with the latest version of the default branch at the time of this bug report.

Exact compiler versions and various paths can be found at the top of the poudriere build log mentioned.

## How to reproduce

* Get a FreeBSD 13.0 amd64 VM image
* Install all required dependencies
* Try to build BambooTracker

(*Describing how to setup a FreeBSD build sandbox would go beyond the scope*)

## System Information

- **Operating System**: FreeBSD 13.0
- **BambooTracker Version**: [v0.5.0](https://github.com/BambooTracker/BambooTracker/releases/download/v0.5.0/BambooTracker-src-v0.5.0.tar.gz)
- **Build Type**: Source tarball with bundled git submodules

**Comment by OPNA2608** [comment]
Can I ask in what QMake project you're doing `qmake` and `make` with? You need to build via the main Project.pro, not BambooTracker/BambooTracker.pro.

BambooTracker/BambooTracker.pro is a sub-project of the main Project.pro. The latter sets up the sub-project dependencies for BambooTracker.pro, including submodules/Rt{Audio,Midi}/Rt{Audio,Midi}.pro - which handle their dependencies and `CONFIG+=system_rt{audio,midi}` checks - and submodules/emu2149/emu2149.pro.

If the Rt\* subprojects are getting hit & you're building against your system Rt\* libraries like your log seems to imply, I'd expect [this message](https://github.com/BambooTracker/BambooTracker/blob/84e61af71c64acdf847dea059266602ba1d109e1/submodules/RtMidi/RtMidi.pro#L4) to get printed, but it's missing from your log. I also don't see [emu2149.c](https://github.com/rerrahkr/emu2149/blob/63c0fe07df89bebdaccbfe982a3f737b39b409d3/emu2149.c) ever getting touched by the compiler. That's why I think you're using the wrong project file.

**Comment by ehaupt** [comment]
Excellent analysis and you're exactly right. The current FreeBSD port ([build recipe](https://cgit.freebsd.org/ports/tree/audio/bambootracker/Makefile)) for version `0.4.5` specified:

```${WRKSRC}``` being the root of the tarball:

```
BUILD_WRKSRC=		${WRKSRC}/BambooTracker
CONFIGURE_WRKSRC=	${WRKSRC}/BambooTracker
```
Therefore it took the (now) wrong project file.

Using the correct project file I was able to build it just fine and I was able to play all the demo files. I expect to be able to update the FreeBSD port soon.

Thank you @OPNA2608.
